Sources/GoogleAI/PartsRepresentable.swift

```diff
@@ -13,117 +13,54 @@
 // limitations under the License.
 
 import Foundation
-import UniformTypeIdentifiers
-#if canImport(UIKit)
-  import UIKit // For UIImage extensions.
-#elseif canImport(AppKit)
-  import AppKit // For NSImage extensions.
-#endif
 
-private let imageCompressionQuality: CGFloat = 0.8
-
-/// A protocol describing any data that could be interpreted as model input data.
+/// A protocol describing any data that could be serialized to model-interpretable input data,
+/// where the serialization process might fail with an error.
 @available(iOS 15.0, macOS 11.0, macCatalyst 15.0, *)
-public protocol PartsRepresentable {
-  var partsValue: [ModelContent.Part] { get }
+public protocol ThrowingPartsRepresentable {
+  func tryPartsValue() throws -> [ModelContent.Part]
 }
 
-/// Enables a `String` to be passed in as ``PartsRepresentable``.
+/// A protocol describing any data that could be serialized to model-interpretable input data,
+/// where the serialization process cannot fail with an error. For a failable conversion, see
+/// ``ThrowingPartsRepresentable``
 @available(iOS 15.0, macOS 11.0, macCatalyst 15.0, *)
-extension String: PartsRepresentable {
-  public var partsValue: [ModelContent.Part] {
-    return [.text(self)]
-  }
+public protocol PartsRepresentable: ThrowingPartsRepresentable {
+  func toPartsValue() -> [ModelContent.Part]
```

Comment:
Not quite sure about the name `toPartsValue`. Shouldn't it be just `partsValue`?

Reply:
Done.